spirv_msl.cpp

```diff
@@ -3793,10 +3807,30 @@ string CompilerMSL::entry_point_args(bool append_comma)
 			auto &m = meta.at(type.self);
 			if (m.members.size() == 0)
 				break;
-			if (!ep_args.empty())
-				ep_args += ", ";
-			ep_args += get_argument_address_space(var) + " " + type_to_glsl(type) + "& " + r.name;
-			ep_args += " [[buffer(" + convert_to_string(r.index) + ")]]";
+			if (!type.array.empty())
+			{
+				// Metal doesn't directly support this, so we must expand the
```

Comment:
Does this apply for array of textures as well? At least MSL 2.0 supports "proper" array of textures.

Reply:
Prior to MSL 2.0 on desktop (1.2 on mobile), yes. (I did mention this possibility in the initial description of this PR.)

Reply:
Oh, ye, now I see.